packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -336,15 +1740,17 @@ class AdapterClass extends EventEmitter {
                 }
             }
             if (this.startedInCompactMode) {
-                this.emit('exit', exitCode, reason);
+                this.emit('exit', _exitCode, reason);
                 adapterStates = null;
                 adapterObjects = null;
             } else {
-                process.exit(exitCode === undefined ? EXIT_CODES.ADAPTER_REQUESTED_TERMINATION : exitCode);
+                process.exit(_exitCode);
             }
         }, 500);
     }
 
+    // external signature
+    getPort(port: number, host?: string, callback?: (port: number) => void): void;
```

Comment:
The overloads should define all possible call signatures, not just one. Here, the case is missing where only port and callback are given.